zerver/tests/test_users.py

```diff
@@ -1595,6 +1627,28 @@ def test_get_user_profile(self) -> None:
         self.assertEqual(result['user']['email'], bot.email)
         self.assertTrue(result['user']['is_bot'])
 
+        # Tests the GET /users/{email} api endpoint.
+        user = self.example_user('hamlet')
+        result = ujson.loads(self.client_get(f'/json/users/{user.email}').content)
+        self.assertEqual(result['user']['email'], user.email)
+        self.assertEqual(result['user']['full_name'], user.full_name)
+        self.assertIn("user_id", result['user'])
+        self.assertNotIn("profile_data", result['user'])
+        self.assertFalse(result['user']['is_bot'])
+        self.assertFalse(result['user']['is_admin'])
+        self.assertFalse(result['user']['is_owner'])
```

Comment:
I think this block of assertions would be better done by just confirming what's returned exactly equals what we got back from the `user_id` format query above.